Fail
file path is :
projects/deltachat-core/rust/src/mimefactory.rs

output is:


error is :
   Compiling deltachat v1.139.6 (/home/k/ogs_PHD/LLM4SE/2Rust_projects/auto_get_functionPair/projects/deltachat-core/rust)
error[E0425]: cannot find value `DC_NOT_VERIFIED` in this scope
   --> src/mimefactory.rs:516:28
    |
516 |     let mut min_verified = DC_NOT_VERIFIED;
    |                            ^^^^^^^^^^^^^^^ not found in this scope

error[E0422]: cannot find struct, variant or union type `dc_e2ee_helper_t` in this scope
   --> src/mimefactory.rs:520:27
    |
520 |     let mut e2ee_helper = dc_e2ee_helper_t { ..Default::default() };
    |                           ^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `memset` in this scope
   --> src/mimefactory.rs:521:5
    |
521 |     memset(&mut e2ee_helper, 0, sizeof(dc_e2ee_helper_t));
    |     ^^^^^^ not found in this scope
    |
help: consider importing this function
    |
3   + use libc::memset;
    |

error[E0425]: cannot find value `dc_e2ee_helper_t` in this scope
   --> src/mimefactory.rs:521:40
    |
521 |     memset(&mut e2ee_helper, 0, sizeof(dc_e2ee_helper_t));
    |                                        ^^^^^^^^^^^^^^^^ help: a local variable with a similar name exists: `e2ee_helper`

error[E0425]: cannot find value `DC_MF_NOTHING_LOADED` in this scope
   --> src/mimefactory.rs:523:57
    |
523 |     if self.factory.is_null() || self.factory.loaded == DC_MF_NOTHING_LOADED || self.factory.out.is_empty() {
    |                                                         ^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0433]: failed to resolve: use of undeclared type `Error`
   --> src/mimefactory.rs:525:20
    |
525 |         return Err(Error::InvalidUse);
    |                    ^^^^^ use of undeclared type `Error`
    |
help: consider importing one of these items
    |
3   + use anyhow::Error;
    |
3   + use async_imap::error::Error;
    |
3   + use async_native_tls::Error;
    |
3   + use async_smtp::error::Error;
    |
      and 30 other candidates

warning: unused import: `Address`
 --> src/mimefactory.rs:9:20
  |
9 | use lettre_email::{Address, Header, MimeMultipartType, PartBuilder};
  |                    ^^^^^^^
  |
note: the lint level is defined here
 --> src/lib.rs:6:5
  |
6 |     unused,
  |     ^^^^^^
  = note: `#[warn(unused_imports)]` implied by `#[warn(unused)]`

warning: unused import: `crate::e2ee::EncryptHelper`
  --> src/mimefactory.rs:18:5
   |
18 | use crate::e2ee::EncryptHelper;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `crate::ephemeral::Timer as EphemeralTimer`
  --> src/mimefactory.rs:19:5
   |
19 | use crate::ephemeral::Timer as EphemeralTimer;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `create_outgoing_rfc724_mid`
  --> src/mimefactory.rs:31:5
   |
31 |     create_outgoing_rfc724_mid, create_smeared_timestamp, remove_subject_prefix, time,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared type `Error`
   --> src/mimefactory.rs:525:20
    |
525 |         return Err(Error::InvalidUse);
    |                    ^^^^^ use of undeclared type `Error`
    |
help: consider importing one of these items
    |
3   + use anyhow::Error;
    |
3   + use async_imap::error::Error;
    |
3   + use async_native_tls::Error;
    |
3   + use async_smtp::error::Error;
    |
      and 31 other candidates

warning: unused import: `crate::e2ee::EncryptHelper`
  --> src/mimefactory.rs:18:5
   |
18 | use crate::e2ee::EncryptHelper;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> src/lib.rs:6:5
   |
6  |     unused,
   |     ^^^^^^
   = note: `#[warn(unused_imports)]` implied by `#[warn(unused)]`

error[E0425]: cannot find function `sizeof` in this scope
   --> src/mimefactory.rs:521:33
    |
521 |     memset(&mut e2ee_helper, 0, sizeof(dc_e2ee_helper_t));
    |                                 ^^^^^^ not found in this scope

error[E0609]: no field `factory` on type `MimeFactory<'_>`
   --> src/mimefactory.rs:523:13
    |
523 |     if self.factory.is_null() || self.factory.loaded == DC_MF_NOTHING_LOADED || self.factory.out.is_empty() {
    |             ^^^^^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

error[E0609]: no field `factory` on type `MimeFactory<'_>`
   --> src/mimefactory.rs:523:39
    |
523 |     if self.factory.is_null() || self.factory.loaded == DC_MF_NOTHING_LOADED || self.factory.out.is_empty() {
    |                                       ^^^^^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

error[E0609]: no field `factory` on type `MimeFactory<'_>`
   --> src/mimefactory.rs:523:86
    |
523 |     if self.factory.is_null() || self.factory.loaded == DC_MF_NOTHING_LOADED || self.factory.out.is_empty() {
    |                                                                                      ^^^^^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

error[E0609]: no field `factory` on type `MimeFactory<'_>`
   --> src/mimefactory.rs:524:25
    |
524 |         set_error(&self.factory, "Invalid use of mimefactory-object.");
    |                         ^^^^^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

error[E0425]: cannot find function `set_error` in this scope
   --> src/mimefactory.rs:524:9
    |
524 |         set_error(&self.factory, "Invalid use of mimefactory-object.");
    |         ^^^^^^^^^ not found in this scope

error[E0609]: no field `out` on type `MimeFactory<'_>`
   --> src/mimefactory.rs:535:10
    |
535 |     self.out = mmap_string_new("");
    |          ^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

error[E0425]: cannot find function `mmap_string_new` in this scope
   --> src/mimefactory.rs:535:16
    |
535 |     self.out = mmap_string_new("");
    |                ^^^^^^^^^^^^^^^ not found in this scope

error[E0609]: no field `out` on type `MimeFactory<'_>`
   --> src/mimefactory.rs:536:30
    |
536 |     mailmime_write_mem(&self.out, &mut col, message.unwrap());
    |                              ^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

error[E0425]: cannot find function `mailmime_write_mem` in this scope
   --> src/mimefactory.rs:536:5
    |
536 |     mailmime_write_mem(&self.out, &mut col, message.unwrap());
    |     ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0277]: the trait bound `RenderedEmail: std::default::Default` is not satisfied
   --> src/mimefactory.rs:543:26
    |
543 |     Ok(RenderedEmail { ..Default::default() })
    |                          ^^^^^^^^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `RenderedEmail`
    |
help: consider annotating `RenderedEmail` with `#[derive(Default)]`
    |
93  + #[derive(Default)]
94  | pub struct RenderedEmail {
    |

error[E0609]: no field `factory` on type `mimefactory::MimeFactory<'_>`
   --> src/mimefactory.rs:523:13
    |
523 |     if self.factory.is_null() || self.factory.loaded == DC_MF_NOTHING_LOADED || self.factory.out.is_empty() {
    |             ^^^^^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

error[E0609]: no field `factory` on type `mimefactory::MimeFactory<'_>`
   --> src/mimefactory.rs:523:39
    |
523 |     if self.factory.is_null() || self.factory.loaded == DC_MF_NOTHING_LOADED || self.factory.out.is_empty() {
    |                                       ^^^^^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

error[E0609]: no field `factory` on type `mimefactory::MimeFactory<'_>`
   --> src/mimefactory.rs:523:86
    |
523 |     if self.factory.is_null() || self.factory.loaded == DC_MF_NOTHING_LOADED || self.factory.out.is_empty() {
    |                                                                                      ^^^^^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

error[E0609]: no field `factory` on type `mimefactory::MimeFactory<'_>`
   --> src/mimefactory.rs:524:25
    |
524 |         set_error(&self.factory, "Invalid use of mimefactory-object.");
    |                         ^^^^^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

error[E0609]: no field `out` on type `mimefactory::MimeFactory<'_>`
   --> src/mimefactory.rs:535:10
    |
535 |     self.out = mmap_string_new("");
    |          ^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

error[E0609]: no field `out` on type `mimefactory::MimeFactory<'_>`
   --> src/mimefactory.rs:536:30
    |
536 |     mailmime_write_mem(&self.out, &mut col, message.unwrap());
    |                              ^^^ unknown field
    |
    = note: available fields are: `from_addr`, `from_displayname`, `sender_displayname`, `selfstatus`, `recipients` ... and 9 others

Some errors have detailed explanations: E0277, E0422, E0425, E0433, E0609.
For more information about an error, try `rustc --explain E0277`.
warning: `deltachat` (lib) generated 4 warnings
error: could not compile `deltachat` (lib) due to 17 previous errors; 4 warnings emitted
warning: build failed, waiting for other jobs to finish...
warning: `deltachat` (lib test) generated 3 warnings (2 duplicates)
error: could not compile `deltachat` (lib test) due to 17 previous errors; 3 warnings emitted
error: command `/home/k/.rustup/toolchains/1.77.1-x86_64-unknown-linux-gnu/bin/cargo test --no-run --message-format json-render-diagnostics` exited with code 101

